### PR TITLE
Deadship

### PR DESCRIPTION
**Pull request by dmalarev**
added a check for the ship's health reaching 0, the game stopping when it's true as well as a few basic test cases

### CODE REVIEW

**Comment by wouterz** [comment]
Looks good for now, later we might want to show a defeat screen instead of stop the game

also nice work on the tests

**Comment by dmalarev** [comment]
yeah, i just added the app.stop more as a place holder for where it will develop later

**Comment by FlorisXll** [comment]
Correct, this is something I should implement :)

File: src/main/java/pixelperfect/Spaceship.java [hunk 1]
```diff
@@ -49,7 +49,21 @@ public void updateHealth(double delta) {
       this.health = 0;
     }
   }
-
+  /**
+   *
```
Comment (jessetilro) [inline src/main/java/pixelperfect/Spaceship.java]: This javadoc comment misses a description of what the method does.

File: src/main/java/pixelperfect/Game.java [hunk 2]
```diff
@@ -43,6 +43,16 @@ public void simpleInitApp() {
 
     // Set up scenegraph.
   }
+  
+  /**
+   * Stops the game if the ship is dead
+   * 
+   */
+  public void deadShip() {
```
Comment (jessetilro) [inline src/main/java/pixelperfect/Game.java]: This method should be called in the simpleUpdate method, otherwise the game is never stopped.

**Comment by jessetilro** [comment]
I have some issues with the formatting when I check out the code locally, my IDE reformats the code. I believe the indentation might be different from our current standard, could that be the case?

**Comment by jessetilro** [comment]
The CheckStyle static analysis plugin for the Maven reporting process reported 9 warnings. Some of those have been introduced before this PR. Please generate a Maven site and inspect all reports before creating a pull-request.

**Comment by dmalarev** [comment]
All checkstyle warnings should now be fixed

**Comment by jessetilro** [comment]
LGTM

**Comment by FlorisXll** [comment]
LGTM

**Pull request merged**